ProxyAgent/Controllers/StatusController.cs

```diff
@@ -47,6 +47,9 @@ public class StatusController : Controller
                 };
             }
 
+            // See https://azure.microsoft.com/en-us/blog/disabling-arrs-instance-affinity-in-windows-azure-web-sites/
+            this.Response.Headers.Add(Proxy.SESSION_AFFINITY_HEADER, Proxy.SESSION_AFFINITY_HEADER_VALUE);
```

Comment:
Do we need to do this here again? Or can Proxy.cs or some common place where we only need do it once for all the requests.

Reply:
I cleaned up the code a bit. One thing to note, is that this header must be added last, avoiding to overwrite the same header in case it exists already, and before starting to write the response payload.